pkg/ruler/mapper.go

```diff
@@ -34,15 +36,17 @@ func (m *mapper) MapRules(user string, ruleConfigs map[string][]legacy_rulefmt.R
 	filenames := []string{}
 
 	// user rule files will be stored as `/<path>/<userid>/filename`
-	path := m.Path + "/" + user + "/"
+	path := filepath.Join(m.Path, user)
 	err := m.FS.MkdirAll(path, 0777)
 	if err != nil {
 		return false, nil, err
 	}
 
 	// write all rule configs to disk
 	for filename, groups := range ruleConfigs {
-		fullFileName := path + filename
+		// Store the encoded file name to better handle `/` characters
+		encodedFileName := base64.URLEncoding.EncodeToString([]byte(filename))
```

Comment:
The main downside of this approach is that the encoded filename is unreadable and makes debugging harder. An alternative may be using `url.PathEscape()` (as far as `%` character is not a problem). Same examples:

- `Hello` > `Hello`
- `Hello world` > `Hello%20world`
- `a/b` > `a%2Fb`

Reply:
I like this better. This means the existing file names are same as well.

Reply:
This would only work for POSIX, no windows. Will that be an issue?